conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -183,6 +187,20 @@ private static void frameworkException(HttpServerExchange exchange, FrameworkExc
         writeResponse(exchange, Optional.of(ConjureError.fromServiceException(exception)), statusCode);
     }
 
+    private static void deadlineExpiredException(HttpServerExchange exchange, DeadlineExpiredException exception) {
+        // need to further narrow down to external versus internal exception types
+        if (exception instanceof DeadlineExpiredException.External externalException) {
+            log.info("external deadline exceeded", externalException);
+            writeResponse(exchange, Optional.empty(), 400);
```

Comment:
How does the calling client know that this was the result of a provided deadline, and not some other 400?

Reply:
ah, good point - i guess we don't make a distinction currently :(

we could tack on a header, or could consider returning something like [HTTP 417](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/417) though I'm wary of re-using that for this purpose.

Perhaps we should add a response header encoder for these exception types similar to `QosReasons.encodeToResponse`?

Reply:
> EDIT: and place that in deadlines-java, not here

or maybe it's better to house that here; i dunno, i'm not very good at these things. the mapping of exception types to a header hinting to the reason seems unique to this codepath, whereas deadlines-java is primarily concerned with throwing an exception of the right type, at the right time.

Reply:
Ya, I really the idea of encapsulating this within the deadlines library

Reply:
draft of that here: https://github.com/palantir/deadlines-java/pull/30

Reply:
i've updated this PR to incorporate the changes that set a header+status